src/toast/ops/demodulation.py

```diff
@@ -634,7 +687,7 @@ class StokesWeightsDemod(Operator):
     )
 
     weights = Unicode(
-        defaults.weights, help="Observation detdata key for output weights"
+        f"{defaults.weights}_demod", help="Observation detdata key for output weights"
```

Comment:
While testing the demodulation I discovered a case where having the same name as the pre-demodulation weights caused a problem.  Unfortunately I don't recall the details, so I can try switching it back and see what breaks.

Reply:
I tried to reproduce my earlier problem, but could not.  In principle, the previously created stokes weights will get deleted and recreated if the demodulated weights have the same name.  This would only be a problem if you wanted to use the pre-demodulated weights for something later.  I have reverted this change.  However, note that all of our workflow tools currently get the name of the weight directly from the operator, so you could change the name to anything and no downstream operators would be affected.